spec/src/main/asciidoc/overview.asciidoc

```diff
@@ -79,12 +79,52 @@ Instances of `ManagedExecutor` and `ThreadContext` can be injected into CDI bean
     unmanagedCompletionStage.thenApply(threadContext.contextualFunction(function3));
 ----
 
-The container provides default instances of `ManagedExecutor` and `ThreadContext`, as injected above, which are shared within an application with all other injection points for `ManagedExecutor` and `ThreadContext` that do not specify a qualifier. Their configuration is equivalent to `ManagedExecutor.builder().build()` and `ThreadContext.builder().build()`.
+The container allows to inject default instances of `ManagedExecutor` and `ThreadContext` which do not leverave either `@NamedInstance` or `ManagedExecutorConfig`/`ThreadContextConfig` annotations, as shown above.
+These instances are not shared and container creates a new instance for each injection point of `ManagedExecutor` or `ThreadContext`.
+The configuration of these instances is equal to default values present in `@ManagedExecutorConfig` and `ThreadContextConfig`.
 
 === Injection of Configured Instances
 
-TODO write this section after discussion about the Config annotations is resolved.
+The `@ManagedExecutorConfig` and `@ThreadContextConfig` provide the ability to configure the injection points with desired capabilities.
+With just the config annotation in place, every injection point is assigned a new instance.
+
+[source, java]
+----
+    @Inject 
+    @ManagedExecutorConfig(maxAsync = 5, propagated = ThreadContext.SECURITY)
+    ManagedExecutor configuredExecutor;
+    ...
+    CompletableFuture<Long> stage = configuredExecutor
+        .newIncompleteFuture()
+        .thenApply(function)
+        .thenAccept(consumer);
+    stage.completeAsync(supplier);
+----
+
+=== Sharing Configured Instances
+
+Configured instances of `ManagedExecutor` and `ThreadContext` can also be shared so long as they are given a CDI qualifier `@NamedInstance` which assigns them a unique name per configuration.
+Given that one injection point contains a configuration and a name, application can inject the same instance elsewhere using just the CDI qualifier as an identifier.
+In the example below, `executor1` injection point defines a configuration and a name `myExec` for a configured `ManagerExecutor` instance.
```

Comment:
In the example below, the `executor1` injection point defines a configuration and a name, `myExec`, for a configured `ManagerExecutor` instance.